qemu/tests/virtio_console.py

```diff
@@ -1632,23 +1646,49 @@ def test_rw_notconnect_guest():
         guest_worker = qemu_virtio_port.GuestWorker(vm)
         virtio_test.cleanup(vm, guest_worker)
 
+    @error_context.context_aware
     def test_rmmod():
         """
         Remove and load virtio_console kernel module.
         :param cfg: virtio_console_params - which type of virtio port to test
-        :param cfg: virtio_port_spread - how many devices per virt pci (0=all)
         """
-        (vm, guest_worker, port) = virtio_test.get_vm_with_single_port(
-            params.get('virtio_console_params'))
-        guest_worker.cleanup()
+        if params.get('virtio_console_params') == 'serialport':
+            vm = virtio_test.get_vm_with_ports(no_serialports=2)
+            _ports, ports = virtio_test.get_virtio_ports(vm)
+            guest_worker = qemu_virtio_port.GuestWorker(vm)
+        else:
+            (vm, guest_worker, ports) = virtio_test.get_vm_with_single_port(
+                params.get('virtio_console_params'))
+            guest_worker.cleanup()
+
         session = vm.wait_for_login()
-        if session.cmd_status('lsmod | grep virtio_console'):
+        if not session.cmd_output('lsmod | grep virtio_console'):
             test.cancel("virtio_console not loaded, probably "
-                        " not compiled as module. Can't test it.")
-        session.cmd("rmmod -f virtio_console")
+                        "not compiled as module. Can't test it.")
+
+        if params.get('virtio_console_params') == 'serialport':
+            error_context.context("Test basic data transfer.", logging.info)
+            basic_loopback(vm, guest_worker, ports[0], ports[1], True)
+            ports[0].open()
+            ports[1].open()
```

Comment:
What's the reason of openning ports here? I found that `basic_loopback()` will open them
Is it for line 1690?

Reply:
Thanks. Actually t is for polarion case reason. The module can not be removed when port is in use, so as a compromise, removed it when port is open.

Reply:
Line 1690 is inherited from the old code, it restarts a `virtio_console_guest` script and init the `guest_worker` object, to verify if the device virtio console can be well initialized after disable/enable module 'virtio_console'.